scm/src/main/java/com/sandboni/core/scm/resolvers/CliChangeScopeResolver.java

```diff
@@ -0,0 +1,154 @@
+package com.sandboni.core.scm.resolvers;
+
+import com.sandboni.core.scm.resolvers.cli.GitDiffRunner;
+import com.sandboni.core.scm.exception.SourceControlException;
+import com.sandboni.core.scm.exception.SourceControlRuntimeException;
+import com.sandboni.core.scm.proxy.SourceControlFilter;
+import com.sandboni.core.scm.revision.RevisionScope;
+import com.sandboni.core.scm.scope.*;
+import com.sandboni.core.scm.utils.RawUtil;
+import com.sandboni.core.scm.utils.ThrowingConsumer;
+import io.reflectoring.diffparser.api.model.Diff;
+import io.reflectoring.diffparser.api.model.Hunk;
+import io.reflectoring.diffparser.api.model.Line;
+import io.reflectoring.diffparser.api.model.Range;
+import org.eclipse.jgit.diff.RawText;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Repository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public class CliChangeScopeResolver extends JGitChangeScopeResolver implements ChangeScopeResolver {
+
+    final static String NON_EXISTING_FILE = "/dev/null";
+    private final static Function<String, String> trimFileName = t -> t.startsWith("a/") || t.startsWith("b/") ? t.substring(2) : t;
+    private final static Function<RawText, String> rawTextToString = t -> t.getString(0, t.size(), false);
+
+    private static final Logger log = LoggerFactory.getLogger(CliChangeScopeResolver.class);
+
+    public CliChangeScopeResolver(Repository repository, SourceControlFilter... filters) {
+        super(repository, filters);
+    }
+
+    private static <T> Consumer<T> throwingConsumerWrapper(ThrowingConsumer<T, Exception> throwingConsumer) {
+        return i -> {
+            try {
+                throwingConsumer.accept(i);
+            } catch (Exception ex) {
+                throw new SourceControlRuntimeException("Unable to process diff", ex);
+            }
+        };
+    }
+
+    @Override
+    public ChangeScope<Change> getChangeScope(RevisionScope<ObjectId> revisionScope) throws SourceControlException {
+        final ChangeScope<Change> scope = new ChangeScopeImpl();
+
+        List<Diff> diffs = GitDiffRunner.diff(repository, revisionScope);
+
+        ObjectId toRevision = revisionScope.getTo();
+
+        diffs.forEach(throwingConsumerWrapper(d -> getChangesForDiff(d, scope, toRevision)));
+
+        return scope;
+    }
+
+
+    private void getChangesForDiff(Diff diff, ChangeScope<Change> scope, ObjectId toRevision) throws IOException {
+
+        String fromFileName = trim(diff.getFromFileName());
+        String toFileName = trim(diff.getToFileName());
+
+        log.debug("processing diff: from file: {} ,to file {}", fromFileName, toFileName);
+
+        if (fileDeleted(toFileName)) {
```

Comment:
To make this method more readable, can we move this line up and do the all logic if `!fileDeleted(toFileName)` ?

Reply:
It is as up as it could be.. I will change to !fileDeleted(toFileName).